packages/synapse-types/src/Chat.ts

```diff
@@ -120,3 +122,40 @@ export type SessionHistoryResponse = {
   page: Interaction[] // Array of interactions in the session's history, ordered by interaction timestamp
   nextPageToken?: string // Optional token to retrieve the next page of results
 }
+
+// TraceEvent type
+export type TraceEvent = {
+  /**
+   * The time stamp identifies when the agent generated this trace event.
+   * It is also used to uniquely identify this event within the context of this asynchronous job.
+   */
+  timestamp: number
+  /**
+   * The trace text message generated by the agent while processing a chat request.
+   */
+  message: string
+}
+
+// TraceEventsResponse type
+export type TraceEventsResponse = {
+  /**
+   * The job ID issued when the agent chat request job was started.
+   */
+  jobId: string
+  /**
+   * A single page of trace events ordered by time stamp descending.
+   */
+  page: TraceEvent[]
+}
+
+export type TraceEventsRequest = {
+  /**
+   * The job ID issued when the agent chat request job was started.
+   */
+  jobId: string
+
+  /**
+   * When a time stamp value is provided, only trace events that occurred after the provided time stamp will be included in the results.
+   */
+  olderThanTimestamp?: number // Optional since the JSON definition allows for it to be omitted
```

Comment:
I commented on John's doc but also calling out here I think this should be `newerThanTimestamp` or something like that

Reply:
Nice catch.  I would expect it to be named newerThanTimestamp (if my understanding about the purpose of this field is correct)